modules/dnn/src/caffe/caffe_importer.cpp

```diff
@@ -569,45 +742,59 @@ class CaffeImporter
 
 }
 
-Net readNetFromCaffe(const String &prototxt, const String &caffeModel /*= String()*/)
+Net readNetFromCaffe(const String &prototxt,
+                     const String &caffeModel, /*= String()*/
+                     int engine)
 {
+    static const int engine_forced = (int)utils::getConfigurationParameterSizeT("OPENCV_FORCE_DNN_ENGINE", ENGINE_AUTO);
+    if(engine_forced != ENGINE_AUTO)
+        engine = engine_forced;
+
     CaffeImporter caffeImporter(prototxt.c_str(), caffeModel.c_str());
     Net net;
-    caffeImporter.populateNet(net);
+    caffeImporter.populateNet(net, engine == ENGINE_NEW || engine == ENGINE_AUTO);
```

Comment:
What about fallback, if the new engine cannot handle the model?

Reply:
Look like it is unnecessary. The parser doesn't fall in tests